features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/internal/vitals/JankStatsActivityLifecycleListener.kt

```diff
@@ -200,20 +173,90 @@ internal class JankStatsActivityLifecycleListener(
         activeActivities[window] = list
     }
 
-    @RequiresApi(Build.VERSION_CODES.N)
+    @MainThread
+    private fun trackWindowJankStats(window: Window) {
+        val knownJankStats = activeWindowsListener[window]
+        if (knownJankStats != null) {
+            internalLogger.log(
+                InternalLogger.Level.DEBUG,
+                InternalLogger.Target.MAINTAINER,
+                { "Resuming jankStats for window $window" }
+            )
+            knownJankStats.isTrackingEnabled = true
+        } else {
+            internalLogger.log(
+                InternalLogger.Level.DEBUG,
+                InternalLogger.Target.MAINTAINER,
+                { "Starting jankStats for window $window" }
+            )
+            val jankStats = jankStatsProvider.createJankStatsAndTrack(window, this, internalLogger)
+            if (jankStats == null) {
+                internalLogger.log(
+                    InternalLogger.Level.WARN,
+                    InternalLogger.Target.MAINTAINER,
+                    { "Unable to create JankStats" }
+                )
+            } else {
+                activeWindowsListener[window] = jankStats
+            }
+        }
+    }
+
+    @SuppressLint("NewApi")
+    @MainThread
+    private fun trackWindowMetrics(isKnownWindow: Boolean, window: Window, activity: Activity) {
+        if (buildSdkVersionProvider.version() >= Build.VERSION_CODES.S && !isKnownWindow) {
+            registerMetricListener(window)
+        } else if (display == null && buildSdkVersionProvider.version() == Build.VERSION_CODES.R) {
+            // Fallback - Android 30 allows apps to not run at a fixed 60hz, but didn't yet have
+            // Frame Metrics callbacks available
+            val displayManager = activity.getSystemService(Context.DISPLAY_SERVICE) as DisplayManager
+            display = displayManager.getDisplay(Display.DEFAULT_DISPLAY)
+        }
+    }
+
+    @RequiresApi(Build.VERSION_CODES.S)
     private fun registerMetricListener(window: Window) {
-        if (buildSdkVersionProvider.version() >= Build.VERSION_CODES.S) {
-            if (frameMetricsListener == null) {
-                frameMetricsListener = DDFrameMetricsListener()
+        if (frameMetricsListener == null) {
+            frameMetricsListener = DDFrameMetricsListener()
+        }
+        val handler = Handler(Looper.getMainLooper())
+        // Only hardware accelerated views can be tracked with metrics listener
+        if (window.peekDecorView()?.isHardwareAccelerated == true) {
+            frameMetricsListener?.let { listener ->
```

Comment:
you are already initializing this above if it is null ...is this really needed here ?